src/burnchains/bitcoin/bits.rs

```diff
@@ -17,33 +17,33 @@
  along with Blockstack. If not, see <http://www.gnu.org/licenses/>.
 */
 
-use bitcoin::util::address as btc_address;
 use bitcoin::blockdata::opcodes::All as btc_opcodes;
 use bitcoin::blockdata::opcodes::Class;
-use bitcoin::blockdata::script::{Script, Instruction, Instructions, Builder};
+use bitcoin::blockdata::script::{Script, Instruction, Builder};
 use bitcoin::blockdata::transaction::TxIn as BtcTxIn;
 use bitcoin::blockdata::transaction::TxOut as BtcTxOut;
 
-use bitcoin::network::message as btc_message;
+use bitcoin::util::hash::Sha256dHash;
 
 use burnchains::{
     BurnchainTxInput, 
     BurnchainTxOutput,
     BurnchainInputType,
     PublicKey,
-    Hash160,
+    BurnchainHeaderHash
 };
 
 use burnchains::bitcoin::Error as btc_error;
 use burnchains::bitcoin::keys::BitcoinPublicKey;
 use burnchains::bitcoin::address::{BitcoinAddress, BitcoinAddressType};
-use burnchains::bitcoin::indexer::BitcoinNetworkType;
-
-use util::hash::to_hex;
+use burnchains::bitcoin::BitcoinNetworkType;
 
 use crypto::sha2::Sha256;
 use crypto::digest::Digest;
 
+use util::log;
+use util::hash::Hash160;
+
```

Comment:
Looking at this code, I think `BurnChainTxInput` as it is currently probably shouldn't be a generic type. Most of its functions are very bitcoin specific (and they accept Bitcoin specific arguments).

Reply:
Sorry -- ignore.